src/ZenithAngleCalc.jl

```diff
@@ -204,103 +28,88 @@ function instantaneous_zenith_angle(date::DateTime,
+    η = η_UTC + λ
 
-    # zenith angle
+    # zenith angle, radians (3.18)
     sza = acos(cos(ϕ)*cos(δ)*cos(η) + sin(ϕ)*sin(δ))
 
     # Set to a max of 90 deg.
     if sza > π/2.0
         sza = π/2.0
     end
 
-    return sza, d
-end
+    # solar azimuth angle
+    azi = atan(sin(η), cos(η)*sin(ϕ) - tan(δ)*cos(ϕ))
 
-"""
-    daily_zenith_angle(date::DateTime,
-                       latitude::FT) where {FT <: Real}
-
-returns the daily averaged zenith angle and earth-sun distance
-at a particular latitude on the given date
-"""
-function daily_zenith_angle(date::DateTime,
-                            latitude::FT) where {FT <: Real}
-    ϕ = deg2rad(latitude)
+    # earth-sun distance, m (3.1)
+    d = astro_unit() * (1.0 - e^2) / (1.0 + e*cos(TA))
 
-    TL = true_longitude(date)
-    TA = true_anomaly(date)
-    ecc = eccentricity(date)
-    γ = obliquity(date)
-    d = earth_sun_dist(ecc, TA)
-
-    # declination, radians
-    δ = mod(asin(sin(γ) * sin(TL)), 2*π)
-    
-    # sunrise/sunset angle
-    T = tan(ϕ) * tan(δ)
-    if T >= 1
-        ηd = π
-    elseif T <= -1
-        ηd = 0.0
-    else
-        ηd = acos(-1 * T)
-    end
-    
-    # daily averaged zenith angle
-    szabar = acos((1/π)*(ηd*sin(ϕ)*sin(δ) + cos(ϕ)*cos(δ)*sin(ηd)))
-
-    return szabar, d
+    return sza, azi, d
 end
 
 """
-    daily_zenith_angle(days_since_equinox::I,
+    daily_zenith_angle(date::DateTime,
+                       latitude::FT,
                        obliquity::FT,
                        perihelion::FT,
-                       eccentricity::FT,
-                       latitude::FT) where {FT <: Real, I <: Int}
-
+                       eccentricity::FT) where {FT <: Real}
 returns the daily averaged zenith angle and earth-sun distance
-at a particular latitude given the days since vernal equinox (defined as March 21),
-orbital obliquity, longitude of perihelion, and eccentricity
+at a particular latitude given the date, orbital obliquity, 
+longitude of perihelion, and eccentricity
 """
-function daily_zenith_angle(days_since_equinox::I,
+function daily_zenith_angle(date::DateTime,
+                            latitude::FT,
                             obliquity::FT,
                             perihelion::FT,
-                            eccentricity::FT,
-                            latitude::FT) where {FT <: Real, I <: Int}
+                            eccentricity::FT) where {FT <: Real}
+    ϕ = deg2rad(latitude)
     γ = deg2rad(obliquity)
     ϖ = deg2rad(perihelion)
     ecc = eccentricity
-    ϕ = deg2rad(latitude)
 
-    # calculate mean anomaly at vernal equinox and mean anomaly
-    β = (1 - ecc^2)^0.5
-    MA_VE = mod(-ϖ + (ecc + ecc^3/4)*(1+β)*sin(ϖ), 2*π)
-    MA = mod(2*π*days_since_equinox / (year_anom() / day_length()) + MA_VE, 2*π)
+    # mean anomaly at vernal equinox, radians (3.11)
+    β = (1-e^2)^(0.5)
+    MAve = -ϖ + (e + e^3/4)*(1+β)*sin(ϖ) - 0.5*e^2*(0.5+β)*sin(2*ϖ) + 0.25*e^3*(1/3+β)*sin(3*ϖ)
```

Comment:
This appears for the second time. Better to make it a function to only have the formula stated once.